src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -541,35 +541,38 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 		alreadyResolved = alreadyResolved ?? new Map<string, string>();
 		const promises: Promise<ITaskSummary>[] = [];
 		if (task.configurationProperties.dependsOn) {
-			for (const dependency of task.configurationProperties.dependsOn) {
-				const dependencyTask = await resolver.resolve(dependency.uri, dependency.task!);
-				if (dependencyTask) {
-					this._adoptConfigurationForDependencyTask(dependencyTask, task);
-					const key = dependencyTask.getMapKey();
-					let promise = this._activeTasks[key] ? this._getDependencyPromise(this._activeTasks[key]) : undefined;
-					if (!promise) {
-						this._fireTaskEvent(TaskEvent.create(TaskEventKind.DependsOnStarted, task));
-						encounteredDependencies.add(task.getCommonTaskId());
-						promise = this._executeDependencyTask(dependencyTask, resolver, trigger, encounteredDependencies, alreadyResolved);
-					}
-					promises.push(promise);
-					if (task.configurationProperties.dependsOrder === DependsOrder.sequence) {
-						const promiseResult = await promise;
-						if (promiseResult.exitCode === 0) {
-							promise = Promise.resolve(promiseResult);
-						} else {
-							promise = Promise.reject(promiseResult);
-							break;
+			if (!this._terminalForTask) {
```

Comment:
@Tyriar I spent most of today trying to get `waitOnExit` to work for restored task terminals to no avail. This line seems suspicious to me. I don't see the waitOnExit ever on the `peristentTerminalProcess`

Reply:
further debugged and it's defined in the local pty service but not the pty service. I'm wondering if it's because it's a function and needs to be a string or other type?

Reply:
This makes sense, this should probably be restored upon reconnect as functions can't/shouldn't be serialized. To do this we will need to expose it on `ITerminalInstance` as I don't think it can be changed currently.

We can still persist `waitOnExit` in the terminal, just not when it's a function.